### PR TITLE
shutils/common.sh: don't require license when build_style=meta

### PR DESCRIPTION
**Pull request by maxice8**
metapackages have no content to be licensed under, there is no reason
to have a license when you aren't actually providing anything for the
users.

### CODE REVIEW

File: common/xbps-src/shutils/common.sh [hunk 1]
```diff
@@ -350,7 +350,13 @@ setup_pkg() {
     fi
 
     # Check if required vars weren't set.
-    _vars="pkgname version short_desc revision homepage license"
+    _vars="pkgname version short_desc revision homepage"
+
+    # Only require license to be set when building metapackages
+    if [ "$build_style" != "meta" ]; then
```
Comment (Vaelatern) [inline common/xbps-src/shutils/common.sh]: Can we also do a check to confirm no files are provided?

**Comment by maxice8** [comment]
On Thu, Nov 15, 2018 at 03:59:54PM +0000, Toyam Cox wrote:
> Vaelatern commented on this pull request.
> 
> 
> 
> > @@ -350,7 +350,13 @@ setup_pkg() {
>      fi
>  
>      # Check if required vars weren't set.
> -    _vars="pkgname version short_desc revision homepage license"
> +    _vars="pkgname version short_desc revision homepage"
> +
> +    # Only require license to be set when building metapackages
> +    if [ "$build_style" != "meta" ]; then
> 
> Can we also do a check to confirm no files are provided?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4763#pullrequestreview-175424291

Then that would be much later problably in one of the hooks.

thing is only build_style=meta can have an empty DESTDIR so if you have build_style=meta in your template then that is what you want.

File: common/xbps-src/shutils/common.sh (hunk 1 above)
Comment (maxice8) [reply common/xbps-src/shutils/common.sh]: That will problably go on common/hooks/pre-pkg/99-pkglint,sh

but there already is a check for empty DESTDIR while not being build_style=meta https://github.com/void-linux/void-packages/blob/master/common/hooks/pre-pkg/99-pkglint.sh#L36

**Comment by Vaelatern** [comment]
So why can't we have a special license to set "metapackage".
Others in the team definitely would be happier with this.

**Comment by Vaelatern** [comment]
No, some meta packages have files. It's easier to document a need to have the license variable specially set than have exceptions that might not catch something that needs a license.

**Comment by maxice8** [comment]
On Thu, Nov 15, 2018 at 08:18:41AM -0800, Toyam Cox wrote:
> So why can't we have a special license to set "metapackage".
> Others in the team definitely would be happier with this.
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4763#issuecomment-439098043

I'm fine with it, just have it documented somewhere.

**Comment by Vaelatern** [comment]
I appreciate that. Sometimes the documentation is the best place!

**Pull request closed without merging**